### PR TITLE
ENH Auto-scaffold blog categories and tags formfields

### PR DESCRIPTION
**Pull request by GuySartorelli**
See post-implementation note in the linked issue about these - short version is that this mirrors the implementation used in `BlogPost`, but I can't update `BlogPost` to take advantage of it just yet. That'll be handled in a separate issue.

## Issue
- https://github.com/silverstripe/.github/issues/262

### CODE REVIEW

**Comment by GuySartorelli** [comment]
JS CI failure is unrelated to these changes

File: src/Model/BlogCategory.php [hunk 1]
```diff
@@ -55,6 +58,44 @@ class BlogCategory extends DataObject implements CategorisationObject
...
+        return $this->scaffoldFormFieldForManyRelation($relationName, $fieldTitle, $ownerRecord);
+    }
+
+    private function scaffoldFormFieldForManyRelation(
+        string $relationName,
+        ?string $fieldTitle,
+        DataObject $ownerRecord
+    ): FormField {
+        $parent = ($ownerRecord instanceof SiteTree) ? $ownerRecord->Parent() : null;
+        $field = TagField::create(
```
Comment (emteknetnz) [inline src/Model/BlogCategory.php:87]: Is there any reason to use TagField over SearchableMultiDropdownField?  From memory their UI is quite similar, though TagField could create (update?) new records from within the field (add a string, press enter, create/update a record), while SearchableMultiDropdownField is for linking existing records which I think is what we're doing here

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: src/Model/BlogCategory.php (hunk 1 above)
Comment (ChloeSartorelli) [reply src/Model/BlogCategory.php:87]: TagField is what it's already using so I'm just copying that logic across.
But yes, the main difference is the ability to create new tags/categories directly in the formfield itself, which makes it quicker to add them.

I'm not aiming to change the way people add or create tags or categories with this PR.

**Review by emteknetnz — APPROVED** [review]
Tested locally, looks good

**Pull request merged**